cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -147,21 +148,47 @@ writeProtocolParameters mOutFile pparams =
         LBS.writeFile fpath (encodePretty pparams)
 
 runQueryTip
-  :: AnyConsensusModeParams
+  :: AnyCardanoEra
+  -> AnyConsensusModeParams
   -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
-runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
+runQueryTip (AnyCardanoEra era) (AnyConsensusModeParams cModeParams) network mOutFile = do
     SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
     let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
-
     tip <- liftIO $ getLocalChainTip localNodeConnInfo
 
-    let output = encodePretty tip
+    let consensusMode = consensusModeOnly cModeParams
+
+    mEpoch <-
+      case toEraInMode era consensusMode of
+        Nothing -> return Nothing
+        Just eraInMode ->
+          case cardanoEraStyle era of
+            LegacyByronEra -> return Nothing
+            ShelleyBasedEra sbe -> do
+              let epochQuery = QueryInEra eraInMode $ QueryInShelleyBasedEra sbe QueryEpoch
+                  cPoint = chainTipToChainPoint tip
+              eResult <- liftIO $ queryNodeLocalState localNodeConnInfo cPoint epochQuery
+              case eResult of
+                Left _acqFail -> return Nothing
+                Right eNum -> case eNum of
+                                Left _eraMismatch -> return Nothing
+                                Right epochNum -> return $ Just epochNum
+
+    let pTip = encodePretty tip
+        epoch = encodePretty $ toObject mEpoch
+        output = LBS.concat [pTip, LBS.fromStrict (Text.encodeUtf8 "\n"), epoch]
```

Comment:
How about merging these into one json object?
```
output = encodePretty (tip <> toObject mEpoch)
```

One object as output is probably what people want anyway.